packages/react-components/react-utilities/src/compose/resolveShorthand.ts

```diff
@@ -26,11 +27,17 @@ export const resolveShorthand: ResolveShorthandFunction = (value, options) => {
 
   let resolvedShorthand = {} as UnknownSlotProps;
 
-  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value) || isValidElement(value)) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value) || isValidElement<any>(value)) {
     resolvedShorthand.children = value;
   } else if (typeof value === 'object') {
     resolvedShorthand = { ...value };
   }
 
-  return defaultProps ? { ...defaultProps, ...resolvedShorthand } : resolvedShorthand;
+  return {
+    ...defaultProps,
+    ...resolvedShorthand,
+    [SLOT_EXTERNAL_CHILDREN_SYMBOL]: resolvedShorthand.children,
```

Comment:
I'm trying to understand why `SLOT_EXTERNAL_CHILDREN_SYMBOL` is needed. What's the difference between that, and `slot.children`?

Reply:
I could see a reason to have `SLOT_RENDER_FUNCTION_SYMBOL` as a symbol instead.

In the case that `value.children` is a function, `resolveShorthand` would:
* Set the function as `SLOT_RENDER_FUNCTION_SYMBOL`.
* _Remove_`slot.children`.

That would make it so this would work correctly, in the case that the user passed a render function for `button.children`:

```jsx
const [slots, slotProps] = getSlotsNext(state);

return (
  <slots.root {...slotProps.root}>
    <slots.button {...slotProps.button}>
      <slots.icon {...slotProps.icon} />
      {slots.button.children /* 💥 this could be a render function */ }
    </slots.button>
  </slots.root>
);
```

Reply:
In fact, if you used `SLOT_RENDER_FUNCTION_SYMBOL`,  you also wouldn't need `SLOT_INTERNAL_CHILDREN_SYMBOL` either.

You only need to keep the default children around in case the user passed in a render function as the child. But if you saved the render function as `SLOT_RENDER_FUNCTION_SYMBOL`, then the default children could just be saved on `slot.children` instead of a custom symbol:

```tsx
  const slot = {
    ...defaultProps,
    ...resolvedShorthand,
  };

  if (typeof slot.children === 'function')
  {
    slot[SLOT_RENDER_FUNCTION_SYMBOL] = slot.children;
    slot.children = defaultProps?.children;
  }

  return slot;
```

Reply:
> I'm trying to understand why `SLOT_EXTERNAL_CHILDREN_SYMBOL` is needed. What's the difference between that, and `slot.children`?

@behowell, in [Problem 1](https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/slot-children-render-function.md#problem-1) we have a conflict between external children and internal children, which one of them gets lost in the process that is solved with `SLOT_INTERNAL_CHILDREN_SYMBOL`. But in [Problem 2](https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/slot-children-render-function.md#problem-2) we have a conflict between whoever wins on the `Problem 1` with an override children, to ensure we'll have the tree instances of children we need 2 variables to maintain reference for those children to be used together with overrides.

Reply:
> In fact, if you used `SLOT_RENDER_FUNCTION_SYMBOL`, you also wouldn't need `SLOT_INTERNAL_CHILDREN_SYMBOL` either.
> 
> You only need to keep the default children around in case the user passed in a render function as the child. But if you saved the render function as `SLOT_RENDER_FUNCTION_SYMBOL`, then the default children could just be saved on `slot.children` instead of a custom symbol:
> 
> ```tsx
>   const slot = {
>     ...defaultProps,
>     ...resolvedShorthand,
>   };
> 
>   if (typeof slot.children === 'function')
>   {
>     slot[SLOT_RENDER_FUNCTION_SYMBOL] = slot.children;
>     slot.children = defaultProps?.children;
>   }
> 
>   return slot;
> ```

Unfortunately this doesn't work on [Problem 2](https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/convergence/slot-children-render-function.md#problem-2), as any reference to `slot.children` will be overridden on render time and we'll lose information, in this particular case we'll lose internal children information.

Reply:
Just a recap here. After better investigation, it's not a big deal to lose the internal children reference in render time, as the priority would be the override children nonetheless. I ended up implementing the suggestions here proposed 👍🏻